trade-finance-manager-ui/server/middleware/validatePostAddPaymentRequestBody/index.ts

```diff
@@ -0,0 +1,73 @@
+import { CURRENCY, Currency, FEE_RECORD_STATUS, FeeRecordStatus } from '@ukef/dtfs2-common';
+import { NextFunction, Request, Response } from 'express';
+import { asUserSession } from '../../helpers/express-session';
+import { AddPaymentErrorKey } from '../../controllers/utilisation-reports/helpers';
+
+const CURRENCY_REGEX_GROUP = `(?<currency>${Object.values(CURRENCY).join('|')})`;
+const CURRENCY_REGEX = new RegExp(CURRENCY_REGEX_GROUP);
+
+const FEE_RECORD_STATUS_REGEX_GROUP = `(?<status>${Object.values(FEE_RECORD_STATUS).join('|')})`;
+const FEE_RECORD_STATUS_REGEX = new RegExp(FEE_RECORD_STATUS_REGEX_GROUP);
+
+const FEE_RECORD_CHECKBOX_KEY_REGEX = new RegExp(`feeRecordId-\\d+-reportedPaymentsCurrency-${CURRENCY_REGEX_GROUP}-status-${FEE_RECORD_STATUS_REGEX_GROUP}`);
+
+const isRequestBodyAnObject = (body: unknown): body is object => !body || typeof body === 'object';
+
+const redirectWithError = (
+  req: Request,
+  res: Response,
+  reportId: string,
+  addPaymentError: AddPaymentErrorKey,
+  checkedCheckboxIds: Record<string, true | undefined> = {},
+) => {
+  req.session.addPaymentErrorKey = addPaymentError;
+  req.session.checkedCheckboxIds = checkedCheckboxIds;
+  return res.redirect(`/utilisation-reports/${reportId}`);
+};
+
+export const validatePostAddPaymentRequestBody = (req: Request, res: Response, next: NextFunction) => {
```

Comment:
This is quite a long function so I am going to see if I can find a way to extract some parts of it.